protocol/messenger.go

```diff
@@ -6260,3 +6274,66 @@ func (m *Messenger) getMessagesToDelete(message *common.Message, chatID string)
 	}
 	return messagesToDelete, nil
 }
+
+func (m *Messenger) withChatClock(callback func(string, uint64) error) error {
+	clock, chat := m.getLastClockWithRelatedChat()
+	err := callback(chat.ID, clock)
+	if err != nil {
+		return err
+	}
+	chat.LastClockValue = clock
+	return m.saveChat(chat)
+}
+
+func (m *Messenger) syncSocialSettings(ctx context.Context, rawMessageHandler RawMessageHandler) error {
```

Comment:
It was not clear to me from the code itself how the messages are dispatched, please consider this small readability improvement.
```suggestion
func (m *Messenger) syncSocialSettings(ctx context.Context, rawMessageDispatcher RawMessageHandler) error {
```

Reply:
fixed @osmaczko